pallets/api/src/messaging/mod.rs

```diff
@@ -471,41 +558,35 @@ pub mod pallet {
 }
 impl<T: Config> Pallet<T> {
 	// Attempt to notify via callback.
-	fn call(
-		origin: AccountIdOf<T>,
+	pub(crate) fn call(
+		origin: &AccountIdOf<T>,
 		callback: Callback<T::AccountId>,
-		id: MessageId,
+		id: &MessageId,
 		data: &impl Encode,
-		deposit: BalanceOf<T>,
 	) -> DispatchResult {
 		let result = T::CallbackExecutor::execute(
 			origin.clone(),
 			[callback.selector.to_vec(), (id, data).encode()].concat(),
```

Comment:
just a comment, but in this spot the `(id,data).encode()` would change depending on what the ABI is.